spec/support/service_template_helper.rb

```diff
@@ -7,8 +7,10 @@ def build_service_template_tree(hash)
   def build_all_atomics(hash)
     hash.each do |name, value|
       next unless value[:type] == "atomic"
-      item  = FactoryGirl.create(:service_template, :name         => name,
-                                                    :service_type => 'atomic')
+      item = FactoryGirl.create(:service_template, :name         => name,
+                                                   :service_type => 'atomic')
+      item.update_attributes(:prov_type => value[:prov_type]) if !value[:prov_type].nil?
```

Comment:
You should be able to use `if value[:prov_type].present?` unless you really need to handle `nil` and not just a blank value.  Otherwise, you want: `if value[:prov_type]`